satellite/satellitedb/bandwidthagreement.go

```diff
@@ -24,36 +24,34 @@ func (b *bandwidthagreement) CreateAgreement(ctx context.Context, agreement *bwa
 	return err
 }
 
-func (b *bandwidthagreement) GetAgreements(ctx context.Context) ([]*bwagreement.Agreement, error) {
+func (b *bandwidthagreement) GetAgreements(ctx context.Context) ([]bwagreement.Agreement, error) {
 	rows, err := b.db.All_Bwagreement(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	var agreements []*bwagreement.Agreement
-	for _, entry := range rows {
-		agreements = append(agreements, &bwagreement.Agreement{
-			Signature: entry.Signature,
-			Agreement: entry.Data,
-			CreatedAt: entry.CreatedAt,
-		})
+	agreements := make([]bwagreement.Agreement, len(rows))
+	for i, entry := range rows {
+		agreement := agreements[i]
+		agreement.Signature = entry.Signature
+		agreement.Agreement = entry.Data
+		agreement.CreatedAt = entry.CreatedAt
 	}
 	return agreements, nil
 }
 
-func (b *bandwidthagreement) GetAgreementsSince(ctx context.Context, since time.Time) ([]*bwagreement.Agreement, error) {
+func (b *bandwidthagreement) GetAgreementsSince(ctx context.Context, since time.Time) ([]bwagreement.Agreement, error) {
 	rows, err := b.db.All_Bwagreement_By_CreatedAt_Greater(ctx, dbx.Bwagreement_CreatedAt(since))
 	if err != nil {
 		return nil, err
 	}
 
-	var agreements []*bwagreement.Agreement
-	for _, entry := range rows {
-		agreements = append(agreements, &bwagreement.Agreement{
-			Signature: entry.Signature,
-			Agreement: entry.Data,
-			CreatedAt: entry.CreatedAt,
-		})
+	agreements := make([]bwagreement.Agreement, len(rows))
+	for i, entry := range rows {
+		agreement := agreements[i]
```

Comment:
You need to take the pointer, otherwise you won't change the actual value in the array.

Reply:
done